test/hotspot/jtreg/runtime/cds/appcds/SharedArchiveConsistency.java

```diff
@@ -146,201 +81,102 @@ private static long getRandomBetween(long start, long end) throws Exception {
-            used[i] = get_region_used_size_aligned(fc, i);
+            used[i] = CDSArchiveUtils.usedRegionSizeAligned(jsaFile, i);
             System.out.printf("%-24s%12d\n", shared_region_name[i], used[i]);
             total += used[i];
         }
         System.out.printf("%-24s%12d\n", "Total: ", total);
-        long header_size = getFileHeaderSize(fc);
-        long region_start_offset = header_size;
-        for (int i=0; i<region; i++) {
-            region_start_offset += used[i];
+        long regionStartOffset = CDSArchiveUtils.fileHeaderSize();
+        for (int i = 0; i < region; i++) {
+            regionStartOffset += used[i];
         }
         if (used[region] == 0) {
             System.out.println("Region " + shared_region_name[region] + " is empty. Nothing to corrupt.");
             return false;
         }
-        System.out.println("Corrupt " + shared_region_name[region] + " section, start = " + region_start_offset
-                           + " (header_size + 0x" + Long.toHexString(region_start_offset-header_size) + ")");
-        long bytes_written = 0L;
-        while (bytes_written < used[region]) {
-            writeData(fc, region_start_offset + bytes_written, bbuf);
-            bbuf.clear();
-            bytes_written += 4096;
-        }
-        if (fc.isOpen()) {
-            fc.close();
+        System.out.println("Corrupt " + shared_region_name[region] + " section, start = " + regionStartOffset
+                           + " (header_size + 0x" + Long.toHexString(regionStartOffset - CDSArchiveUtils.fileHeaderSize()) + ")");
+        long bytesWritten = 0L;
+        while (bytesWritten < used[region]) {
+            CDSArchiveUtils.writeData(jsaFile, regionStartOffset + bytesWritten, buf);
+            bytesWritten += 4096;
         }
+
         return true;
     }
 
     public static void modifyJsaHeader(File jsaFile) throws Exception {
-        FileChannel fc = getFileChannel(jsaFile);
         // screw up header info
-        byte[] buf = new byte[getFileHeaderSize(fc)];
-        ByteBuffer bbuf = ByteBuffer.wrap(buf);
-        writeData(fc, 0L, bbuf);
-        if (fc.isOpen()) {
-            fc.close();
-        }
+        byte[] buf = new byte[CDSArchiveUtils.fileHeaderSize()];
+        System.out.println("CDSArchiveUtils.fileHeaderSize = " + CDSArchiveUtils.fileHeaderSize());
+        CDSArchiveUtils.writeData(jsaFile, 0, buf);
     }
 
     public static void modifyJvmIdent(File jsaFile) throws Exception {
-        FileChannel fc = getFileChannel(jsaFile);
-        int headerSize = getFileHeaderSize(fc);
-        System.out.println("    offset_jvm_ident " + offset_jvm_ident);
-        byte[] buf = new byte[256];
-        ByteBuffer bbuf = ByteBuffer.wrap(buf);
-        writeData(fc, (long)offset_jvm_ident, bbuf);
-        if (fc.isOpen()) {
-            fc.close();
-        }
+        System.out.println("    offset_jvm_ident " + CDSArchiveUtils.offsetJvmIdent);
```

Comment:
I think the modifyXXX functions should also be moved to CDSArchiveUtils so that they can be called by other test cases.

Reply:
Let me double check this part.